src/entities/trades/uniswap-v2/pairs/pairs.ts

```diff
@@ -161,3 +122,57 @@ export async function getAllCommonPairs({
     return list
   }, [])
 }
+
+/**
+ *
+ */
+export async function getAllCommonUniswapV2PairsFromSubgraph({
+  currencyA,
+  currencyB,
+  platform,
+}: GetAllCommonUniswapV2PairsFromSubgraphParams): Promise<Pair[]> {
+  const chainId = (currencyA as Token).chainId ?? (currencyB as Token).chainId
+  const subgraphEndpoint = platform.subgraphEndpoint[chainId]
+
+  if (!subgraphEndpoint) {
+    throw new Error(`No subgraph endpoint for chainId ${chainId}`)
+  }
+
+  const results = await getSdk(new GraphQLClient(subgraphEndpoint)).GetAllCommonPairsBetweenTokenAAndTokenB({
+    tokenA: currencyA.address?.toLowerCase() as string,
+    tokenB: currencyB.address?.toLowerCase() as string,
+  })
+
+  const pairListWithDuplicates = [...results.directPairs, ...results.pairsWithTokenA, ...results.pairsWithTokenB]
+
+  // Remove duplicate pairs
+  const pairList = pairListWithDuplicates.reduce((list, pair) => {
```

Comment:
Could be a bit more easy on the eyes/brain by using a `filter` instead of reducing??

Reply:
Refactored in [7ea391c](https://github.com/levelkdev/swapr-sdk/pull/91/commits/7ea391c9f238d509b95c7b1308377c69cfd259bf)